src/test/regress/input/multi_behavioral_analytics_create_table.source

```diff
@@ -41,3 +41,328 @@ CREATE INDEX is_index4 ON events_table(event_type);
+	LANGUAGE 'internal'
+	AS 'record_lt'
+	IMMUTABLE
+	RETURNS NULL ON NULL INPUT;
+$f$);
+
+SELECT run_command_on_master_and_workers($f$
+
+	CREATE FUNCTION le_user_composite_type_function(user_composite_type, user_composite_type) RETURNS boolean
+	LANGUAGE 'internal'
+	AS 'record_lt'
+	IMMUTABLE
+	RETURNS NULL ON NULL INPUT;
+$f$);
+
+SELECT run_command_on_master_and_workers($f$
+
+	CREATE OPERATOR > (
+	    LEFTARG = user_composite_type,
+	    RIGHTARG = user_composite_type,
+	    PROCEDURE = gt_user_composite_type_function
+	);
+$f$);
+
+SELECT run_command_on_master_and_workers($f$
+
+	CREATE OPERATOR >= (
+	    LEFTARG = user_composite_type,
+	    RIGHTARG = user_composite_type,
+	    PROCEDURE = ge_user_composite_type_function
+	);
+$f$);
+
+-- ... use that function to create a custom equality operator...
+SELECT run_command_on_master_and_workers($f$
+
+	-- ... use that function to create a custom equality operator...
+	CREATE OPERATOR = (
+	    LEFTARG = user_composite_type,
+	    RIGHTARG = user_composite_type,
+	    PROCEDURE = equal_user_composite_type_function,
+		commutator = =, 
+		RESTRICT = eqsel, 
+		JOIN = eqjoinsel, 
+		merges, 
+		hashes
+	);
+$f$);
+
+SELECT run_command_on_master_and_workers($f$
+
+	CREATE OPERATOR <= (
+	    LEFTARG = user_composite_type,
+	    RIGHTARG = user_composite_type,
+	    PROCEDURE = le_user_composite_type_function
+	);
+$f$);
+
+SELECT run_command_on_master_and_workers($f$
+
+	CREATE OPERATOR < (
+	    LEFTARG = user_composite_type,
+	    RIGHTARG = user_composite_type,
+	    PROCEDURE = lt_user_composite_type_function
+	);
+$f$);
+
+
+-- ... and create a custom operator family for hash indexes...
+SELECT run_command_on_master_and_workers($f$
+
+	CREATE OPERATOR FAMILY cats_2_op_fam USING hash;
+$f$);
+
+
+-- ... create a test HASH function. Though it is a poor hash function,
+-- it is acceptable for our tests
+SELECT run_command_on_master_and_workers($f$
+
+	CREATE FUNCTION test_composite_type_hash(user_composite_type) RETURNS int
+	AS 'SELECT hashtext( ($1.tenant_id + $1.tenant_id)::text);'
+	LANGUAGE SQL
+	IMMUTABLE
+	RETURNS NULL ON NULL INPUT;
+$f$);
+
+
+-- We need to define two different operator classes for the composite types
+-- One uses BTREE the other uses HASH
+SELECT run_command_on_master_and_workers($f$
+
+	CREATE OPERATOR CLASS cats_2_op_fam_clas3
+	DEFAULT FOR TYPE user_composite_type USING BTREE AS
+	OPERATOR 1 <=  (user_composite_type, user_composite_type),
+	OPERATOR 2 <  (user_composite_type, user_composite_type),
+	OPERATOR 3 = (user_composite_type, user_composite_type),
+	OPERATOR 4 >= (user_composite_type, user_composite_type),
+	OPERATOR 5 > (user_composite_type, user_composite_type),
+	
+	FUNCTION 1 cmp_user_composite_type_function(user_composite_type, user_composite_type);
+$f$);
+
+SELECT run_command_on_master_and_workers($f$
+	
+	CREATE OPERATOR CLASS cats_2_op_fam_class
```

Comment:
shall we worry about dropping these types, functions, and operators after subquery tests ?